components/elm/src/biogeophys/SoilTemperatureMod.F90

```diff
@@ -943,14 +948,49 @@ subroutine SoilThermProp (bounds,  num_nolakec, filter_nolakec, &
                   endif
                endif
             endif
+            
+            if (use_T_rho_dependent_snowthk) then ! chose which snow thermal conductivity to use 
+               if (snl(c)+1 < 1 .AND. (j >= snl(c)+1) .AND. (j <= 0)) then
+                    bw(c,j) = (h2osoi_ice(c,j)+h2osoi_liq(c,j))/(frac_sno(c)*dz(c,j))
+
+                       do i = 1, 5
```

Comment:
It would be better practice to have the coefficients in arrays and then index those arrays using `i` instead of having a separate case for each value of `i`.